build2cmake/src/config/v2.rs

```diff
@@ -51,6 +53,30 @@ pub struct Torch {
     pub src: Vec<PathBuf>,
 }
 
+#[derive(Debug, Deserialize, Clone, Serialize)]
+#[serde(deny_unknown_fields, rename_all = "kebab-case")]
+pub struct Test {
+    #[serde(default)]
+    pub python_packages: Vec<String>,
```

Comment:
I really like this idea!

Since this is only used in Nix-land, I think it would be nicer to pass optional Python test dependencies as an argument to `genFlakeOutputs`? Then also no special handling is needed for git packages, since it's Nix and you could plug arbitrary derivations in there.

I think `build.toml` isn't the best place, since it suggests generic Python test dependencies and there can be mismatches between Nix Python attribute names and PyPI package names, so they are not generally usable.